app/lib/tool/test_profile/importer.dart

```diff
@@ -0,0 +1,145 @@
+        ..websiteUrl = 'https://${p.name}/'
+        ..created = DateTime.now().toUtc()
+        ..updated = DateTime.now().toUtc()
+        ..isAbandoned = false,
+      ...p.members.map(
+        (m) => PublisherMember()
+          ..parentKey = dbService.emptyKey.append(Publisher, id: p.name)
+          ..id = _userIdFromEmail(m.email)
+          ..userId = _userIdFromEmail(m.email)
+          ..created = DateTime.now().toUtc()
+          ..updated = DateTime.now().toUtc()
+          ..role = PublisherMemberRole.admin,
+      )
+    ]);
+  }
+
+  // create packages
+  Client client;
+  final archiveCacheDir = Directory(archiveCachePath);
+  await archiveCacheDir.create(recursive: true);
+  for (final rv in resolvedVersions) {
+    final parts = rv.split(':');
+    final packageName = parts.first;
+    final versionName = parts[1];
+
+    final fileName = rv.replaceAll(':', '-');
+    final file = File(p.join(archiveCacheDir.path, fileName));
+    // download package archive if not already in the cache
+    if (!await file.exists()) {
+      client ??= Client();
+      final rs = await client.get(
+          '${urls.siteRoot}${urls.pkgArchiveDownloadUrl(packageName, versionName)}');
+      await file.writeAsBytes(rs.bodyBytes);
+    }
+
+    // figure out the active user
+    final testPackage =
+        profile.packages.firstWhere((p) => p.name == packageName);
+    final uploaderEmails = testPackage.publisher == null
+        ? testPackage.uploaders
+        : profile.publishers
+            .firstWhere((p) => p.name == testPackage.publisher)
+            .members
+            .map((m) => m.email)
+            .toList();
+    final uploaderEmail =
+        uploaderEmails[rv.hashCode.abs() % uploaderEmails.length];
+    final activeUser =
+        await accountBackend.lookupUserById(_userIdFromEmail(uploaderEmail));
+
+    // upload package in the name of the active user
+    await fork(() async {
+      registerAuthenticatedUser(activeUser);
+      // ignore: invalid_use_of_visible_for_testing_member
+      await packageBackend.upload(file.openRead());
+
+      if (testPackage.publisher != null) {
+        await packageBackend.setPublisher(
+          packageName,
+          PackagePublisherInfo(
+            publisherId: testPackage.publisher,
+          ),
+        );
+      }
+    });
+  }
+  client?.close();
+
+  // create likes
+  for (final u in profile.users) {
+    if (u.likes == null || u.likes.isEmpty) continue;
+    final userId = _userIdFromEmail(u.email);
+    await dbService.commit(inserts: [
+      ...u.likes.map(
+        (p) => Like()
+          ..parentKey = dbService.emptyKey.append(User, id: userId)
+          ..id = p
+          ..created = DateTime.now().toUtc()
+          ..packageName = p,
+      )
+    ]);
+  }
+}
+
+String _baseIdFromEmail(String email) =>
+    email.replaceAll('.', '-dot-').replaceAll('@', '-at-');
+String _userIdFromEmail(String email) => '${_baseIdFromEmail(email)}-uuid';
```

Comment:
What if we hash the email and derive a uuid from the hash?

Reply:
I've added a small hash prefix, it will be `'8157080d-dev-at-example-dot-com'`. We can tweak this later, but I wanted to keep it readable for now, for easier verification.

Reply:
The reason I propose hashing is that I think it's best to test with UUIDs when this is also what we will be using in production.. and we can we derive deterministic UUIDs using a hash of the email.